### PR TITLE
improve action creation

### PR DESCRIPTION
**Pull request by andy-t-wang**
Move action creation to server side. 

External nullifier is now generated server side, removed permission for user to create actions or set external nullifiers. 

In addition set a unique constraint on the external nullifier to prevent being able to set an already existing external nullifier. These are comprised of app_id and action so we will not have any collisions 

closes DEV-1220

### CODE REVIEW

**Review by ph3t — APPROVED** [review]

File: web/scenes/Portal/Teams/TeamId/Apps/AppId/Actions/page/CreateActionModal/server/form-schema.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import * as yup from "yup";
+
+export const createActionSchema = yup.object().shape({
+  name: yup.string().required("Name is required"),
+  description: yup.string().nullable(),
```
Comment (michalstruck) [inline web/scenes/Portal/Teams/TeamId/Apps/AppId/Actions/page/CreateActionModal/server/form-schema.ts:5]: I think the `allowedCommonCharactersRegex` would fit here

File: web/scenes/Portal/Teams/TeamId/Apps/AppId/Actions/page/CreateActionModal/server/form-schema.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import * as yup from "yup";
+
+export const createActionSchema = yup.object().shape({
+  name: yup.string().required("Name is required"),
```
Comment (michalstruck) [inline web/scenes/Portal/Teams/TeamId/Apps/AppId/Actions/page/CreateActionModal/server/form-schema.ts:4]: `allowedTitleCharactersRegex` would be good here

File: web/scenes/Portal/Teams/TeamId/Apps/AppId/Actions/page/CreateActionModal/server/form-schema.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  description: yup.string().nullable(),
+  action: yup.string().required("Action is required"),
+  app_id: yup.string().required("App ID is required"),
+  max_verifications: yup.number().required("Max verifications is required"),
```
Comment (michalstruck) [inline web/scenes/Portal/Teams/TeamId/Apps/AppId/Actions/page/CreateActionModal/server/form-schema.ts:8]: nit, maybe we require unsigned integers here

Comment (andy-t-wang) [reply web/scenes/Portal/Teams/TeamId/Apps/AppId/Actions/page/CreateActionModal/server/form-schema.ts:8]: ah all very good thanks!

**Pull request merged**